pkg/sysregistriesv2/system_registries_v2_test.go

```diff
@@ -485,3 +488,72 @@ insecure = true`)
 	assert.Equal(t, 4, len(registries))
 	assertSearchRegistryURLsEqual(t, []string{"registry.com", "blocked.registry.com", "insecure.registry.com", "untrusted.registry.com"}, registries)
 }
+
+func toNamedRef(t *testing.T, ref string) reference.Named {
+	parsedRef, err := reference.ParseNamed(ref)
+	require.NoError(t, err)
+	return parsedRef
+}
+
+func TestRewriteReferenceSuccess(t *testing.T) {
+	for _, c := range []struct{ inputRef, prefix, endpointURL, expected string }{
+		// Standard use cases
+		{"example.com/image", "example.com", "example.com", "example.com/image"},
+		{"example.com/image:latest", "example.com", "example.com", "example.com/image:latest"},
+		{"example.com:5000/image", "example.com:5000", "example.com:5000", "example.com:5000/image"},
+		{"example.com:5000/image:latest", "example.com:5000", "example.com:5000", "example.com:5000/image:latest"},
+		{"example.com:5000/image:latest", "", "other.com", "example.com:5000/image:latest"},
+		{"example.com:5000/image:latest", "", "", "example.com:5000/image:latest"},
+		{"example.com:5000/image:latest", "example.com", "example.com", "example.com:5000/image:latest"},
```

Comment:
Per `refMatchesPrefix`, matching `example.com:5000` to `example.com` is unintended/undocumented.